Sources/StreamChat/ChatClient+UserSession.swift

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+extension ChatClient {
+    /// An object which stores shared instances valid during an user session.
+    ///
+    /// Used for types what get expensive when created often.
+    ///
+    /// For example, creating a current user controller for reading a single
+    /// property is expensive if this code path is called often. This is because
+    /// creating a controller leads to database fetch of the model.
+    ///
+    /// ```swift
+    /// if client.currentUserController().currentUser?.privacySettings { …
+    /// ```
+    ///
+    final class UserSession {
```

Comment:
Option 2: No `UserSession` class and put everything into `ChatClient` directly. Maybe too over-engineered based on the single need.

WDYT? This or option 2?

Reply:
I would prefer option 2 tbh, but up to you.

Reply:
I would actually go to option 3 😄  Which is just keep a ref of the current user controller in the `ChatChannelController`

Reply:
Removed this class and kept it more simple by moving it to the `ChatClient` instead.